test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresUnreferencedCodeCapability.cs

```diff
@@ -783,5 +784,182 @@ public static void Test ()
 				Expression<Func<bool>> getter = () => PropertyWithLdToken;
 			}
 		}
+
+		class RequiresOnClass
+		{
+			[RequiresUnreferencedCode ("Message for --ClassWithRequiresUnreferencedCode--")]
+			class ClassWithRequiresUnreferencedCode
+			{
+				public static object Instance;
+
+				public ClassWithRequiresUnreferencedCode () { }
+
+				public static void StaticMethod () { }
+
+				public void NonStaticMethod () { }
+
+				public class NestedClass
+				{
+					public static void NestedStaticMethod () { }
```

Comment:
Is there a test where the nested method inside a nested class calls something with RUC?